Godeps/_workspace/src/k8s.io/kubernetes/pkg/client/cache/delta_fifo.go

```diff
@@ -154,10 +156,30 @@ func (f *DeltaFIFO) Update(obj interface{}) error {
 	return f.queueActionLocked(Updated, obj)
 }
 
-// Delete is just like Add, but makes an Deleted Delta.
+// Delete is just like Add, but makes an Deleted Delta. If the item does not
+// already exist, it will be ignored. (It may have already been deleted by a
+// Replace (re-list), for example.
 func (f *DeltaFIFO) Delete(obj interface{}) error {
+	id, err := f.KeyOf(obj)
+	if err != nil {
+		return KeyError{obj, err}
+	}
 	f.lock.Lock()
 	defer f.lock.Unlock()
+	if f.knownObjects == nil {
+		if _, exists := f.items[id]; !exists {
+			// Presumably, this was deleted when a relist happened.
+			// Don't provide a second report of the same deletion.
+			return nil
```

Comment:
the problem is forcing DeltaFIFO to know whether or not the item currently exists.  That was not previously required by it's behavior.  I prefer to be notified at least once, this logic is going to notify me at most once, assuming no race conditions in the maintenance of the existing object list, which as we've discussed, there are.